syncer/online-ddl-tools/online_ddl.go

```diff
@@ -406,27 +406,27 @@ func (r *RealOnlinePlugin) Apply(tctx *tcontext.Context, tables []*filter.Table,
 	case RealTable:
 		if _, ok := stmt.(*ast.RenameTableStmt); ok {
 			if len(tables) != parserpkg.SingleRenameTableNameNum {
-				return nil, "", "", terror.ErrSyncerUnitGhostRenameTableNotValid.Generate()
+				return nil, terror.ErrSyncerUnitGhostRenameTableNotValid.Generate()
 			}
 
 			tp1 := r.TableType(tables[1].Name)
 			if tp1 == TrashTable {
-				return nil, "", "", nil
+				return nil, nil
 			} else if tp1 == GhostTable {
-				return nil, "", "", terror.ErrSyncerUnitGhostRenameToGhostTable.Generate(statement)
+				return nil, terror.ErrSyncerUnitGhostRenameToGhostTable.Generate(statement)
 			}
 		}
-		return []string{statement}, schema, table, nil
```

Comment:
originallt this is `tables[0].Schema, tables[0].Name` so `targetTables` is one length of tables[0], but after this PR it becomes `tableNames[1:]`

Reply:
Due to the first returns `[]string{statement}`, after calling `Apply`, the following check directly returns so that `tables[0].Schema, tables[0].Name` unused.

```go
// ddl.go L224-226
	if len(sqls) == 0 || (len(sqls) == 1 && sqls[0] == sql) {
		return sqls, nil, nil
	}
```

Reply:
our original logic is totally a mess 😂